x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/deserializer.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { produce } from 'immer';
+
+import { SerializedPolicy } from '../../../../common/types';
+
+import { splitSizeAndUnits } from '../../services/policies/policy_serialization';
+
+import { FormInternal } from './components/phases/types';
+
+export const deserializer = (policy: SerializedPolicy): FormInternal =>
+  produce<FormInternal>(
```

Comment:
Great to see immer in action! 👍 I will have to use it too.

Reply:
It is indeed very convenient! I would caution against using it on massive+nested objects. On lower memory environments it can blow the stack because it recurses :(

Reply:
Good to know!